File: oss/src/main/java/com/truenine/component/oss/properties/OssProperties.java
package com.truenine.component.oss.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * oss属性
 *
 * @author TrueNine
 * @since 2022-10-28
 */
@Data
@ConfigurationProperties(prefix = "component.oss")
public class OssProperties {
  private Type type = Type.FILE;


  /**
   * 类型
   *
   * @author TrueNine
   * @date 2022-10-28
   */
  public enum Type {
    /**
     * 内置文件系统
     */
    FILE,
    /**
     * 存入mysql
     */
    MYSQL_DB,
    /**
     * minio
     */
    MINIO,
    /**
     * 阿里云
     */
    ALI_CLOUD_OSS
  }
}

File: oss/shared/src/main/kotlin/net/yan100/compose/oss/properties/OssProperties.kt
package net.yan100.compose.oss.properties

import org.springframework.boot.context.properties.ConfigurationProperties

private const val PREFIX = "compose.oss"

/**
 * oss属性
 *
 * @param baseUrl 连接 url
 * @param exposeBaseUrl 暴露的 baseUrl
 * @param port 连接端口
 * @author TrueNine
 * @since 2022-10-28
 */
@ConfigurationProperties(prefix = PREFIX)
data class OssProperties(
  var baseUrl: String? = null,
  var exposeBaseUrl: String? = null,
  var port: Int? = null,
)
